### PR TITLE
Create new mobile-metrics ping type Closes #113

### LINKED ISSUE #113: Implement mobile-metrics ping type
Original bug: https://github.com/mozilla-mobile/telemetry-android/issues/68

Schema: https://github.com/mozilla-services/mozilla-pipeline-schemas/blob/dev/schemas/telemetry/mobile-metrics/mobile-metrics.1.schema.json

### PR DESCRIPTION
**Pull request by ekager**

### CODE REVIEW

File: components/service/telemetry/src/main/java/org/mozilla/telemetry/metrics/TelemetryMetrics.java [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+import org.mozilla.telemetry.util.StringUtils;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * TelemetryEvent specifies a common events data format, which allows for broader, shared usage of
+ * data processing tools.
+ */
+public class TelemetryMetrics {
```
Comment (pocmo) [inline components/service/telemetry/src/main/java/org/mozilla/telemetry/metrics/TelemetryMetrics.java]: This looks like a clone of the events class?

This document describes how the metrics will look like:
https://docs.google.com/document/d/1q2V45_fyzm_DGaCos34QEQimsodYJMJCq5Ix-ys9KkE/edit

GeckoView will add an API that will give us the data for "histograms", "keyedHistograms", "scalars" and "keyedScalars".

File: components/service/telemetry/src/main/java/org/mozilla/telemetry/measurement/MetricsMeasurement.java [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.OutputStreamWriter;
+
+public class MetricsMeasurement extends TelemetryMeasurement {
```
Comment (pocmo) [inline components/service/telemetry/src/main/java/org/mozilla/telemetry/measurement/MetricsMeasurement.java]: Metrics will work differently. They do not need to be saved to disk. They will be added at the time we create the ping.

File: components/service/telemetry/src/main/java/org/mozilla/telemetry/ping/TelemetryMobileMetricsPingBuilder.java [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+        addMeasurement(new ProcessStartTimestampMeasurement(configuration));
+        addMeasurement(new SequenceMeasurement(configuration, this));
+        addMeasurement(new LocaleMeasurement());
+        addMeasurement(new OperatingSystemMeasurement());
+        addMeasurement(new OperatingSystemVersionMeasurement());
+        addMeasurement(new CreatedTimestampMeasurement());
+        addMeasurement(new TimezoneOffsetMeasurement());
+        addMeasurement(metricsMeasurement = new MetricsMeasurement(configuration));
+    }
```
Comment (pocmo) [inline components/service/telemetry/src/main/java/org/mozilla/telemetry/ping/TelemetryMobileMetricsPingBuilder.java]: Looking at the doc I think we are missing here: "device", "arch", "profileDate" (Also see core ping).

https://docs.google.com/document/d/1q2V45_fyzm_DGaCos34QEQimsodYJMJCq5Ix-ys9KkE/edit

**Comment by ekager** [comment]
I made it more like core ping but still not sure where it should be built/queued in `Telemetry` or which part of the GV API we will need to expose/deal with here.

File: components/service/telemetry/src/main/java/org/mozilla/telemetry/ping/TelemetryMobileMetricsPingBuilder.java [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+        addMeasurement(new OperatingSystemMeasurement());
+        addMeasurement(new OperatingSystemVersionMeasurement());
+        addMeasurement(new CreatedTimestampMeasurement());
+        addMeasurement(new TimezoneOffsetMeasurement());
+        addMeasurement(metricsMeasurement = new MetricsMeasurement(snapshots));
+    }
+
+    public MetricsMeasurement getMetricsMeasurement() {
+        return metricsMeasurement;
+    }
```
Comment (pocmo) [inline components/service/telemetry/src/main/java/org/mozilla/telemetry/ping/TelemetryMobileMetricsPingBuilder.java]: As we pass the JSON Object in via the constructor, we may not need to keep the reference and do not need a getter.

File: components/service/telemetry/src/test/java/org/mozilla/telemetry/ping/TelemetryMobileMetricsPingBuilderTest.java [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+        assertEquals(1L, results.get("seq"));
+
+        assertTrue(results.containsKey("locale"));
+        assertEquals("en-US", results.get("locale"));
+
+        assertTrue(results.containsKey("os"));
+        assertEquals("Android", results.get("os"));
+
+        assertTrue(results.containsKey("osversion"));
+        assertEquals("27", results.get("osversion")); // API 16 is the default used by this Robolectric version
```
Comment (pocmo) [inline components/service/telemetry/src/test/java/org/mozilla/telemetry/ping/TelemetryMobileMetricsPingBuilderTest.java]: comment says 16 but we check 27

**Pull request merged**